lib/asserter.js

```diff
@@ -105,71 +105,76 @@ class Assertion extends TestResultReporter {
   isNotUndefined() {
     this._notUndefinedAssertion(this.translated('be_defined'));
   }
-  
+
   // Null value assertions
-  
+
   isNull() {
     this._nullAssertion(this.translated('be_null'));
   }
-  
+
   isNotNull() {
     this._notNullAssertion(this.translated('be_not_null'));
   }
-  
+
   // Equality assertions
-  
+
   isEqualTo(expected, criteria) {
     this._equalityAssertion(expected, criteria, true);
   }
-  
+
   isNotEqualTo(expected, criteria) {
     this._equalityAssertion(expected, criteria, false);
   }
-  
+
   // Collection assertions
-  
+
   includes(expectedObject, equalityCriteria) {
     const resultIsSuccessful = this._actual.find(element =>
       this._areConsideredEqual(element, expectedObject, equalityCriteria));
     const failureMessage = `${this.translated('include')} ${Utils.prettyPrint(expectedObject)}`;
     this._reportAssertionResult(resultIsSuccessful, failureMessage);
   }
-  
+
   doesNotInclude(expectedObject, equalityCriteria) {
     const resultIsSuccessful = !this._actual.find(element =>
       this._areConsideredEqual(element, expectedObject, equalityCriteria));
     const failureMessage = `${this.translated('not_include')} ${Utils.prettyPrint(expectedObject)}`;
     this._reportAssertionResult(resultIsSuccessful, failureMessage);
   }
-  
+
   includesExactly(...objects) {
     const resultIsSuccessful = this._haveElementsConsideredEqual(this._actual, objects);
     const failureMessage = `${this.translated('include_exactly')} ${Utils.prettyPrint(objects)}`;
     this._reportAssertionResult(resultIsSuccessful, failureMessage);
   }
-  
-  isEmpty() {
-    const resultIsSuccessful = Utils.numberOfElements(this._actual) === 0;
+
+  isEmpty(value) {
+    const setValueWhenUndefined = value || this._actual
+    const isUndefined = typeof value === undefined || this._actual === undefined;
+
+    const resultIsSuccessful = Utils.numberOfElements(setValueWhenUndefined || {}) === 0  && !isUndefined;
```

Comment:
I would not prefer to do the comparison against `undefined` from the scope of this function. I rather delegate it to `Utils` module,  that has a `notNullOrUndefined` method. This has the advantage that your fix will work for both undefined and null

Reply:
I used your function to check this